static/src/javascripts/projects/common/modules/commercial/article-body-adverts.js

```diff
@@ -62,19 +72,29 @@ define([
 
             if (config.page.hasInlineMerchandise) {
                 adNames.unshift(['im', 'im']);
-                insertAdAtP(spacefinder.getParaWithSpace(lenientRules));
-            }
-            insertAdAtP(spacefinder.getParaWithSpace(rules));
 
-            if (detect.isBreakpoint({ max: 'tablet' })) {
-                insertAdAtP(spacefinder.getParaWithSpace(rules));
+                inlineMercPromise = spacefinder.getParaWithSpace(lenientRules).then(function (space) {
+                    return insertAdAtP(space);
+                });
+            } else {
+                inlineMercPromise = Promise.resolve(null);
             }
+
+            return inlineMercPromise.then(function () {
```

Comment:
:worried: Not too easy to follow

How about

```
return Promise.all([
   insertAdWithLenientRules,
   insertAdWithRules,
   insertAdWithTabletBreakpoint
]);
```

and each of those functions does only one thing?
Is this what's happening in that function? I'm not even sure.

And if you change `return false` with `return Promise.reject()` then you can assume that `init` always returns a `Promise` and you don't have to call `Promise.catch` in other files